fastapi_utils/base_route.py

```diff
@@ -0,0 +1,109 @@
+from typing import Dict, Generic, List, TypeVar
+
+from fastapi import Depends, HTTPException
+from pydantic import BaseModel
+from sqlalchemy.orm import Session
+
+from fastapi_utils.crud_base import Base, CRUDBase
+
+ResponseModelType = TypeVar("ResponseModelType", bound=BaseModel)
+ResponseModelManyType = TypeVar("ResponseModelManyType", bound=BaseModel)
+CreateSchemaType = TypeVar("CreateSchemaType", bound=BaseModel)
+UpdateSchemaType = TypeVar("UpdateSchemaType", bound=BaseModel)
+IDType = TypeVar("IDType")
+
+
+def get_filter_fields(self) -> List[str]:
+    """This would need to get overridden for each BaseRoute where the filter fields are defined.
+
+    Returns:
+        List[str] -- List of fields to filter by
+    """
+    return []
+
+
+class BaseRoute(Generic[ResponseModelType, ResponseModelManyType, CreateSchemaType, UpdateSchemaType, IDType]):
+    """A base route that has the basic CRUD endpoints.
+
+    For read_many
+
+    """
+
+    filter_fields: List[str] = Depends(get_filter_fields)
+    crud_base = CRUDBase(Base)  # type: ignore
```

Comment:
I think it should be possible for the user to specify their own base class, not the one provided by the library. Otherwise I think it would be difficult/impossible to override things like the table naming scheme or similar. 

Moreover, it could add some annoyances when using alembic for migrations.

I'm not sure how hard it would be to refactor to allow a user-specified `Base`, but I think we should try to support that if possible.